File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/keyfederation/download/ExposureKeysPayload.java
package uk.nhs.nhsx.keyfederation.download;

import com.fasterxml.jackson.annotation.JsonCreator;
import uk.nhs.nhsx.diagnosiskeyssubmission.model.StoredTemporaryExposureKey;

import java.util.List;

public class ExposureKeysPayload {
    
    public final String origin;
    public final String batchTag;
    public final List<StoredTemporaryExposureKey> temporaryExposureKeys;

    @JsonCreator
    public ExposureKeysPayload(String origin, String batchTag, List<StoredTemporaryExposureKey> temporaryExposureKeys) {
        this.origin = origin;
        this.batchTag = batchTag;
        this.temporaryExposureKeys = temporaryExposureKeys;
    }

}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/keyfederation/download/ExposureKeysPayload.kt
package uk.nhs.nhsx.keyfederation.download

import uk.nhs.nhsx.diagnosiskeyssubmission.model.StoredTemporaryExposureKey
import uk.nhs.nhsx.keyfederation.BatchTag

data class ExposureKeysPayload(
    val origin: String,
    val batchTag: BatchTag,
    val temporaryExposureKeys: List<StoredTemporaryExposureKey>
)
